gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/converter/EnvelopeSchemaConverter.java

```diff
@@ -112,8 +113,14 @@ public String convertSchema(Schema inputSchema, WorkUnitState workUnit) throws S
           workUnit.getProp(PAYLOAD_SCHEMA_ID_FIELD) : DEFAULT_PAYLOAD_SCHEMA_ID_FIELD;
       String payloadField = workUnit.contains(PAYLOAD_FIELD) ?
           workUnit.getProp(PAYLOAD_FIELD) : DEFAULT_PAYLOAD_FIELD;
-      String schemaKey = String.valueOf(inputRecord.get(schemaIdField));
+
+      Optional<Object> schemaIdValue = AvroUtils.getFieldValue(inputRecord, schemaIdField);
```

Comment:
What is the difference between AvroUtils.getFieldValue(inputRecord, schemaIdField) and inputRecord.get(schemaIdField)? Is this only for adding a nullable check?

Reply:
`AvroUtils.getFieldValue` support nested field